nimbus-ui/nimbusui/src/app/components/platform/form/elements/event-propagation.directive.ts

```diff
@@ -51,12 +54,18 @@ import { LoggerService } from './../../../../services/logger.service';
         err => { this.logger.error('Failed to emit click event' + JSON.stringify(err))}
       );
 
-      this.postProcessing = this.pageService.postResponseProcessing$.subscribe(
-        event => {
-                    if(event === this.path) {
+      this.postProcessing = this.pageService.postResponseProcessing$
+      .filter(p=> (p === this.path))
+      .subscribe(event => {
+                  if(this.form) {
+                    if(this.form.valid) {
                       this.srcElement.removeAttribute('disabled');
                     }
-                 },
+                  }
+                  else {
+                    this.srcElement.removeAttribute('disabled');
+                  }
+                },
```

Comment:
This can be simplified to be written as:
```
if(!this.form || (this.form && this.form.valid)) {
    this.srcElement.removeAttribute('disabled');
}
```